src/test/java/com/google/devtools/build/lib/bazel/repository/decompressor/SevenZDecompressorTest.java

```diff
@@ -0,0 +1,142 @@
+    DecompressorDescriptor.Builder descriptorBuilder =
+        archiveDescriptor().createDescriptorBuilder().setPrefix(ROOT_FOLDER_NAME);
+    Path outputDir = decompress(descriptorBuilder.build());
+    Path fileDir = outputDir.getRelative(INNER_FOLDER_NAME);
+
+    List<String> files =
+        fileDir.readdir(Symlinks.NOFOLLOW).stream()
+            .map(Dirent::getName)
+            .collect(Collectors.toList());
+    assertThat(files).contains(REGULAR_FILENAME);
+  }
+
+  /** Test decompressing a .7z with entries being renamed during the extraction process. */
+  @Test
+  public void testDecompressWithRenamedFiles() throws Exception {
+    String innerDirName = ROOT_FOLDER_NAME + "/" + INNER_FOLDER_NAME;
+
+    HashMap<String, String> renameFiles = new HashMap<>();
+    renameFiles.put(innerDirName + "/" + REGULAR_FILENAME, innerDirName + "/renamedFile");
+    DecompressorDescriptor.Builder descriptorBuilder =
+        archiveDescriptor().createDescriptorBuilder().setRenameFiles(renameFiles);
+    Path outputDir = decompress(descriptorBuilder.build());
+
+    Path fileDir = outputDir.getRelative(ROOT_FOLDER_NAME).getRelative(INNER_FOLDER_NAME);
+    List<String> files =
+        fileDir.readdir(Symlinks.NOFOLLOW).stream()
+            .map((Dirent::getName))
+            .collect(Collectors.toList());
+    assertThat(files).contains("renamedFile");
+    assertThat(fileDir.getRelative("renamedFile").getFileSize()).isNotEqualTo(0);
+  }
+
+  /** Test that entry renaming is applied prior to prefix stripping. */
+  @Test
+  public void testDecompressWithRenamedFilesAndPrefix() throws Exception {
+    String innerDirName = ROOT_FOLDER_NAME + "/" + INNER_FOLDER_NAME;
+
+    HashMap<String, String> renameFiles = new HashMap<>();
+    renameFiles.put(innerDirName + "/" + REGULAR_FILENAME, innerDirName + "/renamedFile");
+    DecompressorDescriptor.Builder descriptorBuilder =
+        archiveDescriptor()
+            .createDescriptorBuilder()
+            .setPrefix(ROOT_FOLDER_NAME)
+            .setRenameFiles(renameFiles);
+    Path outputDir = decompress(descriptorBuilder.build());
+
+    Path fileDir = outputDir.getRelative(INNER_FOLDER_NAME);
+    List<String> files =
+        fileDir.readdir(Symlinks.NOFOLLOW).stream()
+            .map((Dirent::getName))
+            .collect(Collectors.toList());
+    assertThat(files).contains("renamedFile");
+    assertThat(fileDir.getRelative("renamedFile").getFileSize()).isNotEqualTo(0);
+  }
+
+  /** Test that Unicode filenames are handled. **/
+  @Test
+  public void testUnicodeFilename() throws Exception {
+    Path outputDir = decompress(archiveDescriptor().createDescriptorBuilder().build());
+
+    Path path = outputDir.getRelative(ROOT_FOLDER_NAME).getRelative(INNER_FOLDER_NAME);
+    List<String> filenames =
+        path.readdir(Symlinks.NOFOLLOW).stream()
+            .map((Dirent::getName))
+            .map(StringEncoding::internalToPlatform)
```

Comment:
```suggestion
            .map(StringEncoding::internalToUnicode)
```
since you are passing the result to the Truth assertion libraries, which expect regular Java Unicode strings

Reply:
ok, reverted this change.